src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1912,14 +1969,23 @@ static int do_list_mapped_devices(const std::string &format, bool pretty_format)
       f->dump_string("image", cfg.imgname);
       f->dump_string("snap", cfg.snapname);
       f->dump_string("device", cfg.devpath);
+      if (show_all)
+        f->dump_string("cookie", cfg.cookie);
       f->close_section();
     } else {
       should_print = true;
       if (cfg.snapname.empty()) {
         cfg.snapname = "-";
       }
-      tbl << cfg.pid << cfg.poolname << cfg.nsname << cfg.imgname
-          << cfg.snapname << cfg.devpath << TextTable::endrow;
+      if (cfg.cookie.empty()) {
+        cfg.cookie = "-";
+      }
```

Comment:
Let's report an empty cookie as empty (i.e. don't munge it to `-`).  If we add support for custom user-provided cookies in the future, special casing `-` would be awkward.

Reply:
sure, I just followed the `snapname` just above.